src/vm/vm_core.rs

```diff
@@ -483,8 +515,13 @@ impl VirtualMachine {
         exec_scopes: &mut ExecutionScopes,
         hint_data_dictionary: &HashMap<usize, Vec<Box<dyn Any>>>,
         constants: &HashMap<String, BigInt>,
+        hooks: &Vec<Hook>,
```

Comment:
Maybe we can put hooks in `exec_scopes` in order not to alter the function signatures

Reply:
done